### PR TITLE
CoolingTower:VariableSpeed:Merkel model crash with failure on air flow ratio

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
The CoolingTower:VariableSpeed:Merkel model reports warnings, as do other models, to aid diagnostics of performance. If other models also report warnings a crash can occur. This change corrects #6874 by allowing the simulation to continue when this type of error occurs.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
@mjwitte here's the concept. It's either a new message or it's not. So basically it's"

    if ( newMessage) {
    } else {
    }

What I didn't understand is what the purpose of this is...

    Array1D_string const MessageSearch(SearchCounts,
                                       {"InterZone Surface Areas",
                                        "CAUTION -- Interzone",
                                        "Node Connection Error",
                                        "InterZone Surface Azimu",
                                        "InterZone Surface Tilts",
                                        "Suspected non-planar",
                                        "Deprecated",
                                        "Floor Tilt=",
                                        "Roof/Ceiling Tilt=",
                                        "View factors not",
                                        "Unbalanced exhaust air",
                                        "Loads Initialization",
                                        "CalcDaylightMapPoints:",
                                        "Zone Air Heat Balance",
                                        "occupant density is ext",
                                        "Temperature (low) out o",
                                        "Temperature (high) out",
                                        "nominally unused",
                                        "InfraredTransparent",
                                        "No reporting elements"});

So I left the index = 0 in the IF statement. This method assumes the message is unique across all calls to ShowRecurring*ErrorAtEnd and I am not sure that's true, so leaving the index = 0 means developers who have done this right (first call with index = 0) will get the expected result. Those who don't means that message will be counted with the previous message with the same string. So, any comments?

**Comment by rraustad** [comment]
Error file diff's for HospitalLowEnergy are interesting, those for RefrigeratedWarehouse are concerning.

File: src/EnergyPlus/DataErrorTracking.hh [hunk 1]
```diff
@@ -120,6 +120,8 @@ namespace DataErrorTracking {
 
     // Types
 
+    struct newMessageData {std::string Message;};
+
```
Comment (rraustad) [inline src/EnergyPlus/DataErrorTracking.hh]: no longer needed

File: tst/EnergyPlus/unit/CMakeLists.txt [hunk 2]
```diff
@@ -162,6 +162,7 @@ set( test_src
   TranspiredCollector.unit.cc
   UnitaryHybridAirConditioner.unit.cc
   UnitHeater.unit.cc
+  UtilityRoutines.unit.cc
   VariableSpeedCoils.unit.cc
```
Comment (rraustad) [inline tst/EnergyPlus/unit/CMakeLists.txt:166]: where's my unit test?

**Comment by nrel-bot-3** [comment]
@mjwitte @rraustad @lgentile it has been 14 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
Confirmed the defect using a transitioned defect file and weather data with the develop branch.  Confirmed this fix allows it to work with this branch.  Code changes are OK.  CI is OK, the error file shows splitting up the error message counters.  I'm merging!

**Pull request merged**